src/components/App.jsx

```diff
@@ -61,39 +75,63 @@ function App() {
     setActiveModal("");
   };
 
-  const handleRegistration = (values) => {
-    return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        if (values.email && values.password && values.userName) {
-          const newUser = registerUser({ email: values.email, password: values.password, userName: values.userName });
-          setCurrentUser(newUser);  
-          setIsLoggedIn(true);
-          resolve();
+  const handleRegistration = (email, password, name) => {
+    return api
+      .register(email, password, name)
+      .then(() => handleOpenSuccessPopup())
+      .catch((error) => {
+        if (error.response && error.response.status === 400) {
+          const backendErrors = error.response.data.details || {};
+          setErrors((prevErrors) => ({
+            ...prevErrors,
+            ...backendErrors,
+          }));
         } else {
-          reject(new Error("Invalid registration data"));
+          console.error("Signup failed:", error);
+          alert("Registration failed. Please try again.");
         }
-      }, 1000);
-    });
+        throw error;
+      });
   };
-
+  
   const handleLogin = (email, password) => {
-    const user = loginUser(email, password);  
-    if (user) {
-      setCurrentUser(user);
-      localStorage.setItem('currentUser', JSON.stringify(user));  
-      setIsLoggedIn(true);
-      setActiveModal("");  
```

Comment:
`setActiveModal("")` in line 139 should be replaced with `handleActiveModalClose()` not to duplicate the code